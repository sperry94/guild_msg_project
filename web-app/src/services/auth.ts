import { createContext, useContext, useState, useEffect } from "react"
import { auth, User } from "firebase/app"
import { firebaseAuth } from "./firebase"

export const AuthContext = createContext<User | null | undefined>(undefined)

export const useAuthContext = () => useContext(AuthContext)

export const useAuthUpdate = () => {
  const [auth, setAuth] = useState<User | null | undefined>(undefined)

  useEffect(() => {
    return firebaseAuth().onAuthStateChanged(setAuth, err => {
      console.error("An error occurred checking auth state", err)
      setAuth(null)
    })
  }, [])

  return auth
}

let innerGoogleAuthProvider: auth.GoogleAuthProvider | null = null

const googleAuthProvider = () =>
  innerGoogleAuthProvider ||
  (innerGoogleAuthProvider = new auth.GoogleAuthProvider())

export const signIn = () =>
  firebaseAuth()
    .signInWithRedirect(googleAuthProvider())
    .catch(err => {
      console.error("An error occurred logging in", err)
    })

export const signOut = () =>
  firebaseAuth()
    .signOut()
    .catch(err => {
      console.error("An error occurred logging out", err)
    })
