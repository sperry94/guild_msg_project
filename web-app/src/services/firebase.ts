import * as firebase from "firebase/app"

import "firebase/auth"

const firebaseConfig = {
  apiKey: "AIzaSyAFkLfldso4y5CJGj8BNCoAv9r7Ukacrow",
  authDomain: "guild-msg-project.firebaseapp.com",
  databaseURL: "https://guild-msg-project.firebaseio.com",
  projectId: "guild-msg-project",
  storageBucket: "guild-msg-project.appspot.com",
  messagingSenderId: "88742455218",
  appId: "1:88742455218:web:602d0164baff3181bd66ab",
  measurementId: "G-WCV6QT9SZ5",
}

let innerFirebaseApp: firebase.app.App | undefined

export const firebaseApp = () =>
  innerFirebaseApp ||
  (innerFirebaseApp = firebase.initializeApp(firebaseConfig))

let innerFirebaseAuth: firebase.auth.Auth | undefined

export const firebaseAuth = () =>
  innerFirebaseAuth || (innerFirebaseAuth = firebaseApp().auth())
