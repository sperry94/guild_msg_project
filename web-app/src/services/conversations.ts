import { useState, useEffect } from "react"
import { firestore, firebaseFunctions } from "./firebase"
import { useAuthContext } from "./auth"

export type User = {
  displayName: string
  email: string
  photoUrl: string
  uid: string
}

type ServerConversation = {
  id: string
  users: string[]
}

export type Conversation = {
  id: string
  users: User[]
}

const cachedUserData: { [k: string]: User } = {}

export const useConversationList = (
  errCb: (msg: string) => void = () => undefined
) => {
  const [conversations, setConversations] = useState<Conversation[] | null>(
    null
  )

  const auth = useAuthContext()

  useEffect(() => {
    if (!auth) {
      setConversations([])
      return
    }

    return firestore()
      .collection("conversations")
      .where("users", "array-contains", auth.uid)
      .onSnapshot(
        async snap => {
          const conversations = snap.docs.map(
            doc =>
              ({
                id: doc.id,
                ...doc.data(),
              } as ServerConversation)
          )

          const allUsers = new Set(
            conversations.flatMap(c => c.users as string[])
          )

          allUsers.delete(auth.uid)

          const userData = await Promise.all(
            Array.from(allUsers)
              .filter(u => cachedUserData[u] === undefined)
              .map(u =>
                firestore()
                  .collection("users")
                  .doc(u)
                  .get()
              )
          )

          for (const u of userData) {
            const uData = u.data() as User
            cachedUserData[uData.uid] = uData
          }

          const convertedConversations: Conversation[] = conversations.map(
            c => ({
              ...c,
              users: c.users
                .filter(u => u !== auth.uid)
                .map(u => cachedUserData[u]),
            })
          )

          setConversations(convertedConversations)
        },
        err => {
          console.error(
            "An error occurred reading the conversations for the current user",
            err
          )
          errCb(
            err?.message ||
              "An error occurred reading the conversations for the current user"
          )
          setConversations([])
        }
      )
  }, [auth, errCb])

  return conversations
}

export const addConversation = async (email: string): Promise<string> => {
  if (!email) {
    throw new Error("An email must be provided to add a new conversation!")
  }

  const res = await firebaseFunctions().httpsCallable("addConversation")({
    email,
  })

  return res.data.docId as string
}
