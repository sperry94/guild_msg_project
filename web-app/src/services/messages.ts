import { useState, useEffect } from "react"
import { firestore as innerFirestore } from "firebase/app"
import { firestore, firebaseAuth } from "./firebase"

export type Message = {
  time: innerFirestore.Timestamp
  senderId: string
  message: string
}

export const useMessages = (conversationId?: string) => {
  const [messages, setMessages] = useState<Message[] | null>(null)

  useEffect(() => {
    if (!conversationId) {
      setMessages([])
      return
    }

    return firestore()
      .collection("conversations")
      .doc(conversationId)
      .collection("messages")
      .orderBy("time")
      .limitToLast(50)
      .onSnapshot(
        snap => {
          setMessages(snap.docs.map(doc => doc.data() as Message))
        },
        err => {
          console.error("An error occurred reading the conversation data", err)
          setMessages([])
        }
      )
  }, [conversationId])

  return messages
}

export const sendMessage = async (conversationId: string, message: string) => {
  if (!conversationId || !message) {
    throw new Error(
      "A conversation ID and a message must be passed in order to save."
    )
  }

  const senderId = firebaseAuth().currentUser?.uid

  if (!senderId) {
    throw new Error("A user must be logged-in in order to save.")
  }

  await firestore()
    .collection("conversations")
    .doc(conversationId)
    .collection("messages")
    .add({
      time: innerFirestore.Timestamp.now(),
      message,
      senderId,
    })
}
