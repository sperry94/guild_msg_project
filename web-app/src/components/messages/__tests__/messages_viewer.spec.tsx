import React from "react"
import { useAuthContext } from "../../../services/auth"
import renderer from "react-test-renderer"
import MessagesViewer, { MessagesViewerProps } from "../messages_viewer"

jest.mock("../../../services/auth")
jest.mock("../../../services/firebase")
jest.mock("../../../services/conversations")
jest.mock("../../../services/messages")

const mocked_useAuthContext = useAuthContext as jest.Mock<any>

describe("MessagesViewer", () => {
  const cases: MessagesViewerProps[] = [
    {
      conversationId: undefined,
    },
    {
      conversationId: "convo-id",
    },
  ]

  cases.forEach((c, index) => {
    it(`renders for case #${index + 1} with auth`, () => {
      mocked_useAuthContext.mockReturnValueOnce({})
      const tree = renderer.create(<MessagesViewer {...c} />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it(`renders for case #${index + 1} without auth`, () => {
      mocked_useAuthContext.mockReturnValueOnce(null)
      const tree = renderer.create(<MessagesViewer {...c} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
