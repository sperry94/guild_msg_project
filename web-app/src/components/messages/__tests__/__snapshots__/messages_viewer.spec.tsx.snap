// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesViewer renders for case #1 with auth 1`] = `
<div
  className="makeStyles-content-1"
>
  <div
    className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-4"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-5 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      style={
        Object {
          "visibility": undefined,
        }
      }
    >
      <div
        className="makeStyles-toolbarOffset-3"
      />
      <div
        className="MuiCircularProgress-root makeStyles-spinner-50 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-addNewBtn-6"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            New Conversation
          </p>
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-activeConversation-2 makeStyles-content-153"
  >
    <p
      className="MuiTypography-root makeStyles-noConversationText-156 MuiTypography-h6"
    >
      Please select a conversation from the list on the left or start a new one!
    </p>
    <div
      className="MuiPaper-root makeStyles-content-158 MuiPaper-elevation5"
    >
      <form
        action="#"
        className="makeStyles-form-159"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-input-160"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              aria-label="Message"
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
              disabled={true}
              name="message"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-202 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-203"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Send
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`MessagesViewer renders for case #1 without auth 1`] = `
<div
  className="makeStyles-content-1"
>
  <div
    className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-4"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-5 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      style={
        Object {
          "visibility": undefined,
        }
      }
    >
      <div
        className="makeStyles-toolbarOffset-3"
      />
      <div
        className="MuiCircularProgress-root makeStyles-spinner-50 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-addNewBtn-6"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            New Conversation
          </p>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    className="makeStyles-activeConversation-2 makeStyles-content-153"
  >
    <div
      className="MuiPaper-root makeStyles-content-158 MuiPaper-elevation5"
    >
      <form
        action="#"
        className="makeStyles-form-159"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-input-160"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              aria-label="Message"
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
              disabled={true}
              name="message"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-202 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-203"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Send
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`MessagesViewer renders for case #2 with auth 1`] = `
<div
  className="makeStyles-content-1"
>
  <div
    className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-4"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-5 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      style={
        Object {
          "visibility": undefined,
        }
      }
    >
      <div
        className="makeStyles-toolbarOffset-3"
      />
      <div
        className="MuiCircularProgress-root makeStyles-spinner-50 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-addNewBtn-6"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            New Conversation
          </p>
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-activeConversation-2 makeStyles-content-153"
  >
    <div
      className="MuiCircularProgress-root makeStyles-spinner-157 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="MuiPaper-root makeStyles-content-158 MuiPaper-elevation5"
    >
      <form
        action="#"
        className="makeStyles-form-159"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-input-160"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              aria-label="Message"
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="message"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-202 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-203"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Send
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`MessagesViewer renders for case #2 without auth 1`] = `
<div
  className="makeStyles-content-1"
>
  <div
    className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-4"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-5 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      style={
        Object {
          "visibility": undefined,
        }
      }
    >
      <div
        className="makeStyles-toolbarOffset-3"
      />
      <div
        className="MuiCircularProgress-root makeStyles-spinner-50 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-addNewBtn-6"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            New Conversation
          </p>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    className="makeStyles-activeConversation-2 makeStyles-content-153"
  >
    <div
      className="MuiPaper-root makeStyles-content-158 MuiPaper-elevation5"
    >
      <form
        action="#"
        className="makeStyles-form-159"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-input-160"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              aria-label="Message"
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="message"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-202 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-203"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Send
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
